src/core/client/stream/tabs/Profile/Settings/DeleteAccount/DeleteAccountModal.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { FunctionComponent, useCallback, useState } from "react";
+
+import { Box, Modal } from "coral-ui/components";
+
+import CompletionPage from "./Pages/CompletionPage";
+import ConfirmPage from "./Pages/ConfirmPage";
+import DescriptionPage from "./Pages/DescriptionPage";
+import DownloadCommentsPage from "./Pages/DownloadCommentsPage";
+import WhenPage from "./Pages/WhenPage";
+
+import styles from "./DeleteAccountModal.css";
+
+interface Props {
+  userID: string;
+  scheduledDeletionDate?: string;
+  organizationEmail: string;
+  open: boolean;
+  onClose: () => void;
+}
+
+const DeleteAccountModal: FunctionComponent<Props> = ({
+  userID,
+  open = false,
+  onClose,
+  scheduledDeletionDate,
+  organizationEmail,
+}) => {
+  const [step, setStep] = useState(0);
+
+  const closeModal = useCallback(() => {
+    setStep(0);
+    onClose();
+  }, [onClose, setStep]);
+  const incrementStep = useCallback(() => {
+    setStep(step + 1);
+  }, [step, setStep, closeModal]);
+
+  return (
+    <>
+      <Modal
+        open={open}
+        onClose={closeModal}
+        data-testid="delete-account-modal"
+      >
+        <Box className={styles.root}>
```

Comment:
Suggest breaking the modal contents into a separate component so it's current step can be stored in that component, and you wouldn't need to set it to `0` when you close it (as the Modal content component wouldn't be mounted when the modal is closed).